### PR TITLE
supplementary order.. not beautified.. but functionally there

### PR DESCRIPTION
**Pull request by AJH693**
There will need to be an update to models.. change log is there with what I think would work

### CODE REVIEW

File: src/changelog.txt [hunk 1]
```diff
@@ -1,3 +1,11 @@
+###April  21st
+Need to run migartion script since added new column to models task_response
+
+alter table task_reponse add supplementaryOrder Text
```
Comment (travisreed-wf) [inline src/changelog.txt]: that should be fine

File: src/taskBuilder/views.py [hunk 2]
```diff
@@ -182,6 +182,9 @@ def post(self, taskID):
             task_response.task_id = int(taskID)
             task_response.student_id = current_user.id
             task_response.supplementary = json.dumps(data.get('supplementary'))
+            task_response.supplementaryOrder = json.dumps(data.get('supplementaryOrder'))
+            print "***********HERE**************************"
```
Comment (travisreed-wf) [inline src/taskBuilder/views.py]: remove prints

File: src/taskBuilder/views.py [hunk 3]
```diff
@@ -182,6 +182,9 @@ def post(self, taskID):
             task_response.task_id = int(taskID)
             task_response.student_id = current_user.id
             task_response.supplementary = json.dumps(data.get('supplementary'))
+            task_response.supplementaryOrder = json.dumps(data.get('supplementaryOrder'))
```
Comment (travisreed-wf) [inline src/taskBuilder/views.py]: underscores would be better in the future for anything that is in python.

File: src/templates/grading/response.html [hunk 4]
```diff
@@ -79,6 +79,14 @@
           </div>
         </div>
       {% endfor %}<br>
+      
+      {% if supplementaryOrder %}  
+      <div id="supplementaryOrder">
+      {% for suppOrder in supplementaryOrder%}
+        {{suppOrder}}<br>
```
Comment (travisreed-wf) [inline src/templates/grading/response.html]: what does this look like?

File: src/templates/tasks/tasks_js.js [hunk 5]
```diff
@@ -116,6 +117,7 @@
     }); 
     var modalID = $(modal).attr('id');
     var startTime = startDate.getTime();
+    supplementartInformationOrder.push($(modal).attr('id'));
```
Comment (travisreed-wf) [inline src/templates/tasks/tasks_js.js]: did you mean to make the variable "supplementaryInformationOrder" with a y?

File: src/changelog.txt [hunk 6]
```diff
@@ -1,3 +1,11 @@
+###April  21st
+Need to run migartion script since added new column to models task_response
```
Comment (aguibert) [inline src/changelog.txt]: what migration script?

**Comment by AJH693** [comment]
screen shot 2015-04-22 at 10 13 49 am
Here is the simplified look

**Comment by AJH693** [comment]
I also forgot that I changed the archive course and registration code to be in the option table in course view
screen shot 2015-04-22 at 10 14 26 am

File: src/templates/grading/response.html (hunk 4 above)
Comment (aguibert) [inline src/templates/grading/response.html]: Put some sort of text label here indicating what this info will be for.  In the screenshot you just have a grouping of IDs but its not really clear what the text is for.

**Comment by travisreed-wf** [comment]
Im wondering if this should go into an integration branch. I could then have access to what I need, and can clean up the view before it goes into master

**Comment by AJH693** [comment]
I don't really think that it's necessary atm. I will be working on beautification next.

**Comment by travisreed-wf** [comment]
If this gets merged in, it will go out to production, but it isn't production ready yet.

I think this should either go into an integration branch, or the order should be moved to actually be under the response.

ID:2
Title: asdf
Time: ...
Expected Time: ...
Order Reviewed: 3


It shouldnt be too hard. You should just be able to cast your string to a list using .split() and then get the indices of the relevant supp_ids

**Comment by AJH693** [comment]
That's not how he wants it though. He wants a list of in what order the SI is reviewed.. so like
133123.. and if we add this to each individual SI, then the author will have to scroll up and down to find out the list. I feel like there should be an easy to view list.

**Comment by travisreed-wf** [comment]
I dont think that is what he wants.

**Comment by travisreed-wf** [comment]
@aguibert how did you interpret the ticket?

**Comment by aguibert** [comment]
I interpreted it like hartman said 1,3,3,1,2,3 (but use commas at least).
Especially since we are only storing 1 view time for each supp material, it wouldn't really make sense to print out full supp info multiple times...
I think this would look confusing:
```
ID:1
Title: asdf
Time: 60s
Expected Time: 30s
Order Reviewed: 1
------------------------------
ID:2
Title: qwer
Time: 60s
Expected Time: 30s
Order Reviewed: 2
-----------------------------
ID:1
Title: asdf
Time: 60s
Expected Time: 30s
Order Reviewed: 3
```
Note that ID1 is reviewed for 60s total, but from looking at this list you see the 60s twice so it seems like they reviewed it for 120s.  
Also, he's going to be doing most of the data analysis from the excel data so that's the important bit we need here.  I think doing your approach travis would be a lot more work.

**Comment by travisreed-wf** [comment]
So, in excel are we going to just include the list, or the rank for each
item?

On Wed, Apr 22, 2015 at 2:32 PM, Andrew Guibert <notifications@github.com>
wrote:

> I interpreted it like hartman said 1,3,3,1,2,3 (but use commas at least).
> Especially since we are only storing 1 view time for each supp material,
> it wouldn't really make sense to print out full supp info multiple times...
> I think this would look confusing:
>
> ID:1
> Title: asdf
> Time: 60s
> Expected Time: 30s
> Order Reviewed: 1
> ------------------------------
> ID:2
> Title: qwer
> Time: 60s
> Expected Time: 30s
> Order Reviewed: 2
> -----------------------------
> ID:1
> Title: asdf
> Time: 60s
> Expected Time: 30s
> Order Reviewed: 3
>
> Note that ID1 is reviewed for 60s total, but from looking at this list you
> see the 60s twice so it seems like they reviewed it for 120s.
>
> Also, he's going to be doing most of the data analysis from the excel data
> so that's the important bit we need here. I think doing your approach
> travis would be a lot more work.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ISUWebPortal/491Project/pull/153#issuecomment-95311791>
> .
>



-- 
*Travis Reed *
*Release Management*
*Workiva formerly WebFilings *

*2900 University Blvd, Ames, IA 50010*

**Comment by aguibert** [comment]
Yes, we will just make another column called "order" or something and it will just have the stringified array.  From what I remember that's what we described on the whiteboard in the meeting and Nir agreed with that.

**Comment by travisreed-wf** [comment]
ok.

+1

On Wed, Apr 22, 2015 at 2:40 PM, Andrew Guibert <notifications@github.com>
wrote:

> Yes, we will just make another column called "order" or something and it
> will just have the stringified array. From what I remember that's what we
> described on the whiteboard in the meeting and Nir agreed with that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ISUWebPortal/491Project/pull/153#issuecomment-95313309>
> .
>



-- 
*Travis Reed *
*Release Management*
*Workiva formerly WebFilings *

*2900 University Blvd, Ames, IA 50010*

**Pull request merged**